flink-table-store-core/src/main/java/org/apache/flink/table/store/table/AbstractFileStoreTable.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.table;
+
+import org.apache.flink.table.store.file.schema.Schema;
+import org.apache.flink.table.types.logical.RowType;
+
+/** Abstract {@link FileStoreTable}. */
+public abstract class AbstractFileStoreTable implements FileStoreTable {
+
+    private static final long serialVersionUID = 1L;
+
+    private final String name;
+    protected final Schema schema;
+
+    public AbstractFileStoreTable(String name, Schema schema) {
+        this.name = name;
+        this.schema = schema;
+    }
+
+    @Override
+    public String name() {
+        return name;
```

Comment:
I'm a little hesitant about this one: 1.
1. this will lead to the schema tied to the name, the rename of the table may become very troublesome in the future. Need to go to modify the name.
2. The name is already saved in the catalog, so saving another copy in the schema is a bit redundant.